platforms/desktop-shared/Makefile.common

```diff
@@ -58,15 +58,26 @@ ifeq ($(UNAME_S), Darwin) #APPLE
 
 	CXXFLAGS += `sdl2-config --cflags`
 	CXXFLAGS += -I/usr/local/include -I/opt/local/include
-	CFLAGS = $(CXXFLAGS)
+	CFLAGS += $(CXXFLAGS)
 endif
 
 ifeq ($(findstring MINGW,$(UNAME_S)),MINGW)
 	ECHO_MESSAGE = "MinGW"
 	LIBS += -lgdi32 -lopengl32 -lglew32 -limm32 `pkg-config --static --libs sdl2`
 
 	CXXFLAGS += `pkg-config --cflags sdl2`
-	CFLAGS = $(CXXFLAGS)
+	CFLAGS += $(CXXFLAGS)
+endif
+
+ifneq ($(findstring $(UNAME_S),BSD DragonFly),,)
```

Comment:
It looks like this conditional is messing other build paths

https://github.com/drhelius/Gearboy/pull/140/checks?check_run_id=2982713297

Reply:
I think I prefer a more generic solution here (really the "BSD" block should work on any generic POSIX-like OS) but my lack of familiarity with GNU make syntax prevented me from implementing one ;)

We've had gearboy and gearsystem in [pkgsrc](https://pkgsrc.org) for a long time now, and the build also gets tested on Solaris/illumos, but that needs further work.